keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ResourceStatusService.kt

```diff
@@ -93,6 +95,12 @@ class ResourceStatusService(
     return first() is ResourceCreated
   }
 
+  private fun List<ResourceHistoryEvent>.isWaiting(): Boolean {
+    // we expect to have only two events, but we will accept several different "unresolvable" events
+    // in order to be less brittle and show the user this status instead of an error
+    return all { it is ResourceCreated || it is ResourceCheckUnresolvable } && size < 5
```

Comment:
Maybe reorder this so you don't actually look at a bunch of events unnecessarily?
```suggestion
    return size < 5 && all { it is ResourceCreated || it is ResourceCheckUnresolvable } 
```